### PR TITLE
#185106761 Setup redux

### PR DESCRIPTION
**Pull request by kirengaboris**
### What does this PR do?
- Setup redux
### Description of Task to be completed?
- Setup redux with the redux toolkit
### How should this be manually tested?
- clone the project and use the terminal to start the development server by "npm run dev"
### Any background context you want to provide?
none
### What are the relevant pivotal tracker/Trello stories?
- Tracker: 185106761
### Screenshots (if appropriate)
- none
### Questions:
- none

### CODE REVIEW

**Review by masterchief-00 — APPROVED** [review]

**Review by dean-daryl — APPROVED** [review]

File: src/components/Counter.jsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  const handleDecrement = () => {
+    dispatch(decrement());
+  };
+  const handleReset = () => {
+    dispatch(reset());
+  };
+  const handleIncrementByValue = (e) => {
+    e.preventDefault();
+    dispatch(incrementByAmount(value));
+  };
```
Comment (fab-ryan) [inline src/components/Counter.jsx]: create a custom hook that will allow us to consume those actions and states then we can call that hook after destructing the function or action we need [reference](https://redux-toolkit.js.org/api/other-exports/#bindactioncreators)

**Review by fab-ryan — CHANGES_REQUESTED** [review]
work on that comment after I will merge it

File: src/components/Counter.jsx (hunk 1 above)
Comment (fab-ryan) [reply src/components/Counter.jsx]: that hook can contain two things reduces, and dispatched actions but they are different hook but in the same file

File: src/hooks/useStore.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  };
+  const handleDecrement = () => {
+    dispatch(decrement());
+  };
+  const handleReset = () => {
+    dispatch(reset());
+  };
+  const handleIncrementByValue = (value) => {
+    dispatch(incrementByAmount(value));
+  };
```
Comment (fab-ryan) [inline src/hooks/useStore.js]: no this is not what I told you 
in this file no dispatch each dispatch everything but it will work if it's called

**Review by fab-ryan — CHANGES_REQUESTED** [review]
do reference I gave you

File: src/hooks/useStore.js (hunk 2 above)
Comment (kirengaboris) [reply src/hooks/useStore.js]: can you elaborate on this please

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**